kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/CheckUtils.kt

```diff
@@ -281,21 +285,13 @@ public suspend fun memberFor(event: Event): MemberBehavior? {
 
         event is MemberJoinEvent -> event.member
         event is MemberUpdateEvent -> event.member
+        event is MessageCreateEvent -> event.member
+        event is MessageDeleteEvent -> event.message?.data?.guildId?.value
+            ?.let { event.kord.unsafe.member(it, event.message!!.data.authorId) }
 
-        event is MessageCreateEvent && event.message.getGuildOrNull() != null ->
-            event.message.getAuthorAsMember()
-
-        event is MessageDeleteEvent && event.message?.getGuildOrNull() != null ->
-            event.message?.getAuthorAsMember()
-
-        event is MessageUpdateEvent && event.message.asMessageOrNull()?.getGuildOrNull() != null ->
-            event.getMessage().getAuthorAsMember()
-
-        event is ReactionAddEvent && event.message.asMessageOrNull()?.getGuildOrNull() != null ->
-            event.getUserAsMember()
-
-        event is ReactionRemoveEvent && event.message.asMessageOrNull()?.getGuildOrNull() != null ->
-            event.getUserAsMember()
+        event is MessageUpdateEvent -> event.message.asMessageOrNull()?.getAuthorAsMember()
```

Comment:
`asMessageOrNull` hits the cache to get the message entity, so you could get a member behavior here instead of going for the full member entity